src/Group_User.php

```diff
@@ -954,8 +954,20 @@ public function post_purgeItem()
 
        // remove user from plannings
         $groups_id  = $this->fields['groups_id'];
+        $users_id = $this->fields['users_id'];
         $planning_k = 'group_' . $groups_id . '_users';
 
+        // If user's default group is affected, remove it from user
+        $user_inst = new User();
+        if ($user_inst->getFromDB($users_id) && $user_inst->fields['groups_id'] == $groups_id) {
+            $user_inst->update(
+                [
+                    'id'        => $users_id,
+                    'groups_id' => 0,
+                ]
+            );
+        }
+
        // find users with the current group in their plannings
         $user_inst = new User();
```

Comment:
Instead of creating a new instance for the second operation, the same instance should be used, reducing redundancy.